src/components/BorrowerProfile/BorrowerSocialShare.vue

```diff
@@ -0,0 +1,293 @@
+		KvIcon
+	},
+	props: {
+		borrowerName: {
+			type: String,
+			default: ''
+		},
+		sector: {
+			type: String,
+			default: ''
+		}
+	},
+	data() {
+		return {
+			copyStatus: {
+				class: '',
+				disabled: false,
+				text: ''
+			},
+			shareMessage: '',
+			shareLink: `https://${this.$appConfig.host}${this.$route.path}`
+		};
+	},
+	computed: {
+		facebookShareUrl() {
+			return this.getFullUrl('https://www.facebook.com/dialog/share', {
+				app_id: this.$appConfig.fbApplicationId,
+				display: 'page',
+				href: `${this.shareLink}?utm_source=facebook.com&utm_medium=social&utm_campaign=social_share_checkout`,
+				redirect_uri: `${this.shareLink}?kiva_transaction_id=${this.$route.query.kiva_transaction_id}`,
+				quote: this.shareMessage,
+			});
+		},
+		emailShareUrl() {
+			return this.getFullUrl('Mailto:', {
+				Subject: `Help fund ${this.borrowerName} 's ${this.sector} loan through Kiva!`,
+				// eslint-disable-next-line max-len
+				Body: `Hello,\n\nI thought you might be interested in supporting this Kiva loan, ${this.shareLink}.\n\n`
+					+ `Even a small amount could help fund ${this.borrowerName}'s loan. And if you can't make a `
+					+ 'donation, it would be great if you could share the fundraiser to help spread the word.'
+					+ '\n\nThanks for taking a look!.',
+			});
+		},
+		twitterShareUrl() {
+			return this.getFullUrl('https://twitter.com/intent/tweet', {
+				text: this.shareMessage,
+				url: `${this.shareLink}?utm_source=t.co&utm_medium=social&utm_campaign=social_share_checkout`,
+				via: 'Kiva',
+			});
+		},
+	},
+	methods: {
+		getFullUrl(base, args) {
+			const querystring = _map(args, (val, key) => `${key}=${encodeURIComponent(val)}`).join('&');
+			return `${base}?${querystring}`;
+		},
+		handleFacebookResponse() {
+			// Check for the route hash that facebook adds to the request
+			if (this.$route.hash === '#_=_') {
+				// Check for an error
+				const { error_code: code, error_message: message } = this.$route.query;
+				if (code) {
+					// The 4201 error code means the user pressed 'Cancel', so can be ignored
+					if (code !== '4201') {
+						this.$showTipMsg(`There was a problem sharing to Facebook: ${message}`, 'warning');
+					}
+				} else {
+					this.$showTipMsg('Thanks for sharing to Facebook!');
+				}
+			}
+		},
+		async copyLink() {
+			const url = `${this.shareLink}?utm_source=social_share_link&utm_campaign=social_share_checkout`;
+			try {
+				await clipboardCopy(url);
+				this.copyStatus = {
+					class: 'social__btn--success',
+					disabled: true,
+					text: 'Copied!'
+				};
+			} catch (err) {
+				this.copyStatus = {
+					class: 'social__btn--error',
+					disabled: true,
+					text: 'Error'
+				};
+			} finally {
+				setTimeout(() => {
+					this.copyStatus = {
+						class: '',
+						disabled: false,
+						text: ''
+					};
+				}, 2500);
+			}
+		}
+	},
+	mounted() {
+		this.handleFacebookResponse();
+	},
+};
+</script>
+
+<style lang="scss" scoped>
```

Comment:
Did most of this come from our existing Social Share component? A little bummed to have so much here that isn't in tailwinds but suppose we can roll with it for experimentation sake.

Reply:
Yeah, I would love to work on that. I think we can do it if everything goes well with the experiment. What do you think @emuvente ?